### ISSUE TITLE
xkeyval Error: `url' undefined

### ISSUE THREAD

**Issue by ziima** [michal-h21/biblatex-iso690#59]
I've tried to switch to ISO690 (`iso-authoryear`), but I can't compile the output because of error:
```
! Package xkeyval Error: `url' undefined in families `blx@opt@pre'.
```
I'm not sure what's wrong, since URL should be supported by ISO690.

Also I see a warning
```
Package biblatex Warning: 'iso8601' date format specifier is deprecated, use 'e
dtf' instead.
```
which may or may not be related.

**Comment by michal-h21** [comment]
I cannot reproduce these issues with sample files distributed with the package. Could you please post a minimal working example that causes a trouble? Which TeX distribution do you use? Have you tried to use the development version of `Biblates-iso690`, or did you use the version provided by the distribution?

**Comment by moewew** [comment]
Without an example I can only guess, but I think that @ziima is trying to use `url` as an option, i.e. something like `\usepackage[style=iso-authoryear, url=false]{biblatex}`. The ISO style does not support `url` as an option (to suppress the URL). The URL field, of course, is still supported and the URLs are shown according to the ISO rules.

The warning is unrelated, but should be taken care of.

**Comment by ziima** [comment]
@moewew was right, the cryptical message really pointed to `url` package option, which is not supported by iso690. The same applies to `doi` and `isbn` options.

Is there a way to make the message more informative about the problem?

**Comment by moewew** [comment]
We can't really change `(x)keyval`'s error messages. What theoretically we could do is to define the option and let it do nothing, or throw an error/warning, but I don't quite like that idea. You get a similar message if you use an obviously useless option like `florbel=foo`.

The documentation should be clear about the supported options. Note that the `biblatex` manual in §3.1.2.2 *Style-specific [options]* states clearly that `url` and friends are not core-options (i.e. supported by the `biblatex` core and thus available in all styles), but options implemented by the standard styles (and as such only guaranteed to exist if you use the standard styles, custom styles might choose not to implement them).

**Comment by ziima** [comment]
+1 if the `biblatex` provides a useful message. But I'm unable to report the issue, as the `biblatex` internals are beyond my insight.

**Comment by moewew** [comment]
Mhhh, the error I get is
```
! Package keyval Error: glrbel undefined.
```
where `glrbel` is the unknown option. I think that is fairly clear.

Since `biblatex` delegates its option parsing to `keyval` it would require a rewrite of `keyval`'s macros to inject a 'better' message.

I'm not sure if that is worth it to get something like
```
! Package biblatex Error: option glrbel undefined.
```

If you think that is worth it, you can start an issue at the `biblatex` bugtracker https://github.com/plk/biblatex/issues

**Comment by DavidLuptak** [comment]
The most of this is related to `biblatex` itself, but regarding the `url` and other options, they are implemented since a881ab6b5a4abff2333a24f14f0f4c767bec64a4.
